cli/internal/codegen/ts.go

```diff
@@ -307,6 +307,15 @@ func (ts *ts) writeDeclDef(ns string, decl *schema.Decl) {
 	ts.WriteString("\n")
 }
 
+func (ts *ts) writeGenericTypes() {
+	ts.WriteString("export interface ErrorResponse {\n")
+	ts.WriteString("    code: number,\n")
+	ts.WriteString("    message: string,\n")
+	ts.WriteString("    details: any\n")
+	ts.WriteString("}\n\n")
+	ts.WriteString("export type Result<T> = { data: T } | { error: ErrorResponse }\n\n")
```

Comment:
Isn't a problem to have to do properties assertion to infer result type ?

Reply:
@alexisvisco It shouldn't be, in TypeScript it is quite easy to it with:
```ts
if('error' in response){
 // handle error case
}
```
Besides I believe that it's a good way to not forget to handle the errors. What do you think?